From: NIIBE Yutaka <gniibe@fsij.org>
Date: Tue, 18 Apr 2017 09:04:11 +0900
Subject: dirmngr: Fix final close of LISTEN_FD.

* dirmngr/dirmngr.c (handle_connections): Close LISTEN_FD.

Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>
(cherry picked from commit 4b2581dc0ea1d03e70023bb0748aa0c21c0a2173)
---
 dirmngr/dirmngr.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/dirmngr/dirmngr.c b/dirmngr/dirmngr.c
index 31d3ca2..513e2a6 100644
--- a/dirmngr/dirmngr.c
+++ b/dirmngr/dirmngr.c
@@ -1905,7 +1905,6 @@ handle_connections (assuan_fd_t listen_fd)
 #endif
   struct sockaddr_un paddr;
   socklen_t plen = sizeof( paddr );
-  gnupg_fd_t fd;
   int nfd, ret;
   fd_set fdset, read_fdset;
   int saved_errno;
@@ -2030,6 +2029,8 @@ handle_connections (assuan_fd_t listen_fd)
 
       if (FD_ISSET (FD2INT (listen_fd), &read_fdset))
 	{
+          gnupg_fd_t fd;
+
           plen = sizeof paddr;
 	  fd = INT2FD (npth_accept (FD2INT(listen_fd),
 				    (struct sockaddr *)&paddr, &plen));
@@ -2058,7 +2059,6 @@ handle_connections (assuan_fd_t listen_fd)
                 }
 	      npth_setname_np (thread, threadname);
             }
-          fd = GNUPG_INVALID_FD;
 	}
     }
 
@@ -2067,8 +2067,8 @@ handle_connections (assuan_fd_t listen_fd)
     close (my_inotify_fd);
 #endif /*HAVE_INOTIFY_INIT*/
   npth_attr_destroy (&tattr);
-  if (listen_fd != -1)
-    assuan_sock_close (fd);
+  if (listen_fd != GNUPG_INVALID_FD)
+    assuan_sock_close (listen_fd);
   cleanup ();
   log_info ("%s %s stopped\n", strusage(11), strusage(13));
 }
